### PR TITLE
add link to token page

### PR DESCRIPTION
**Pull request by 7dir**

### CODE REVIEW

**Comment by whiteinge** [comment]
The link is a good addition, thanks.

Switching ^M to newlines is breaking the tests. (I haven't checked to see if it works with the output from GitHub yet.) What are your thoughts on that change?

I agree with your TODO. I had forgotten I was using hexdump. That isn't in posix and should be replaced.

**Comment by whiteinge** [comment]
Thoughts on #41 to remove hexdump?

**Comment by whiteinge** [comment]
Sorry, I didn't mean for you to close this PR. I'd still like to get your link addition in and I'd like to hear if the awk-based temp_dir change works for you -- I'd prefer not to rely on hexdump or on /dev/urandom.

**Comment by 7dir** [comment]
https://www.cyberciti.biz/tips/shell-scripting-bash-how-to-create-temporary-random-file-name.html

```sh
#So to create a temp file:

tfile=$(mktemp /tmp/foo.XXXXXXXXX)
echo "a file: $tfile"
#To create a temp dir:

tdir=$(mktemp -d /tmp/foo.XXXXXXXXX)
echo "a direcotry: $tdir"
```

tested in windows git bash

**Comment by whiteinge** [comment]
Thanks but `mktemp` (the command, not the C function) is [not specified by POSIX](http://pubs.opengroup.org/onlinepubs/9699919799/idx/im.html) so I'd prefer to avoid it.

**Comment by 7dir** [comment]
https://unix.stackexchange.com/questions/174817/finding-the-correct-tmp-dir-on-multiple-platforms

**Comment by 7dir** [comment]
oh, yeah...

```md
Requirements

A POSIX environment (tested against Busybox v1.19.4)
```

I don`t know

File: ok.sh [hunk 1]
```diff
@@ -690,8 +690,7 @@ _response() {
     _log debug 'Processing response.'
 
     read -r http_version status_code status_text
-    status_text="${status_text%
-}"
+    status_text="${status_text%}"
```
Comment (whiteinge) [inline ok.sh]: The `^M` characters here were intentional. Is there a reason for the removal?

**Comment by whiteinge** [comment]
I'm fairly confident with using awk to generate a random directory name (#41) and we're already restricting perms with the `mkdir -m` below. I think that's the most portable approach.

File: ok.sh (hunk 1 above)
Comment (7dir) [inline ok.sh]: oh
sorry )

**Comment by 7dir** [comment]
|windows bash
I have no success using it (ok.sh) anytime
should I use token istead pass?

```
machine api.github.com
    login 7dir
    password token or pass
```

going use github-api (js)

thanks for talking )

ps
it`s still good idea (sh)
look at this https://github.com/ingydotnet/git-hub

**Comment by whiteinge** [comment]
I haven't tried with a password so I'm not sure if GitHub supports that or not. Probably better to use a token since they're easier to revoke/change.

Thanks for the report, I'll give this a test on Windows. What shell environment are you using?

**Comment by 7dir** [comment]
git bash https://git-scm.com/
windows 7 32 bit

**Comment by whiteinge** [comment]
@7dir thanks for the report! There were a few interesting issues in that environment. I think #43 should address them. I didn't test everything so if you happen to try this again please let me know if you notice any breakage.

**Comment by 7dir** [comment]
great - it works

**Comment by 7dir** [comment]
```sh
./ok.sh create_repo baasd
parse error: Invalid escape at line 1, column 23
jq parse error; invalid JSON.
Client Error: 400 Bad Request
```

**Comment by whiteinge** [comment]
Thanks once again for the report. Two issues: one is fixed in #44 and the other is curl on Windows is expecting the netrc file to be under `C:\Users\<user>\_netrc`. I added a note about that to the README.

I will think on how to get the tests running under Windows. It seems there are several subtle differences.

**Comment by 7dir** [comment]
I can make test!

**Pull request closed without merging**